src/axolotl/utils/models.py

```diff
@@ -394,10 +394,14 @@ def apply_patches(self) -> None:
             and self.cfg.flash_attention
             and self.cfg.sample_packing
         ):
+            has_remote_code = (
+                "auto_map" in self.model_config
+                and self.model_type in self.model_config["auto_map"]
```

Comment:
```suggestion
                and `AutoModel` in self.model_config["auto_map"]
```
I think this is probably what we're looking for then?

Reply:
oh, I thought since self.model_type is `AutoModelForCausalLM` in most cases and we want to look for it. I suppose it is fine to check for either

https://github.com/axolotl-ai-cloud/axolotl/blob/e4af51eb66ffea450bdcd55b10be2b52661ad17e/src/axolotl/utils/models.py#L362

https://github.com/axolotl-ai-cloud/axolotl/blob/e4af51eb66ffea450bdcd55b10be2b52661ad17e/docs/config.qmd#L21

https://github.com/axolotl-ai-cloud/axolotl/blob/e4af51eb66ffea450bdcd55b10be2b52661ad17e/src/axolotl/utils/config/models/input/v0_4_1/__init__.py#L108

Reply:
`self.model_config` contains the values of the dict in `config.json`, typically based on the Model's Config class. so in this case, I think we're looking for `AutoModelForCausalLM` based on https://huggingface.co/microsoft/Phi-3.5-MoE-instruct/blob/main/config.json#L10, and not on the actual model_type defined by the axolotl yaml config.